app/src/main/java/com/example/myapplication/CreateAccountActivity.kt

```diff
@@ -22,31 +27,31 @@ class CreateAccountActivity : AppCompatActivity() {
         nextButton.setOnClickListener {
             val email = emailField.text.toString().trim()
             val password = passwordField.text.toString().trim()
-
-            if (email.isEmpty()) {
-                emailField.error = "Email is required"
-                return@setOnClickListener
-            }
-
-            if (password.isEmpty()) {
-                passwordField.error = "Password is required"
-                return@setOnClickListener
-            }
-
-            if (password.length < 6) {
-                passwordField.error = "Password must be at least 6 characters"
-                return@setOnClickListener
-            }
-
             val rememberMe = rememberMeCheckbox.isChecked
 
-            Toast.makeText(this, "Signed in", Toast.LENGTH_SHORT).show()
+            when {
+                email.isEmpty() -> emailField.error = "Email is required"
```

Comment:
1) use string resources here also, don't hardcode them
https://developer.android.com/guide/topics/resources/string-resource

2) Since you display error in two cases, how about creating a method to do it so it's more readable?
setEmailError(...)